autoload/neomake/signs.vim

```diff
@@ -44,18 +59,17 @@ function! neomake#signs#PlaceSign(entry) abort
     endif
 endfunction
 
-function! neomake#signs#CleanOldSigns() abort
-    if !empty(s:last_placed_signs)
-        call neomake#utils#DebugObject('Cleaning old signs:', s:last_placed_signs)
+function! neomake#signs#CleanOldSigns(bufnr) abort
+    if !has_key(s:last_placed_signs, a:bufnr) || empty(s:last_placed_signs[a:bufnr])
```

Comment:
I think I advised you to move the `empty` call here, but now I think we don't need it at all since we are `unlet`ing at the end of this function call. We'll free a bit more memory without it, too.